### PR TITLE
[AS-902] When user accepts ToS, add user to OpenDJ if they are enabled

### PR DESCRIPTION
**Pull request by jordansafer**
Ticket: https://broadworkbench.atlassian.net/browse/AS-902
- Added logic to enabled OpenDJ when ToS is accepted
- Updated tests to reflect the change

---

**PR checklist**

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've filled out the [Security Risk Assessment](https://broadinstitute.github.io/dsp-appsec-security-risk-assessment/) and attached the result to the JIRA ticket

### CODE REVIEW

**Review by sagehen03 — APPROVED** [review]
lgtm with the caveat that I don't know scala well enough to bless the UserService changes.  From the tests, seems pretty clear that they work, but would be good to get a scala veteran's take on whether they're idiomatic.

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/service/UserService.scala [hunk 1]
```diff
@@ -143,6 +143,9 @@ class UserService(val directoryDAO: DirectoryDAO, val cloudExtensions: CloudExte
       case Some(_) =>
         for {
           _ <- tosService.acceptTosStatus(userId)
+          _ <- directoryDAO.isEnabled(userId, samRequestContext).flatMap(enabled => {
+            if (enabled) registrationDAO.enableIdentity(userId, samRequestContext) else IO.none
+          })
           status <- IO.fromFuture(IO(getUserStatus(userId, false, samRequestContext)))
         } yield status
```
Comment (davidangb) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/service/UserService.scala:149]: for-comprehensions are implemented under the covers as flatMaps, so you could write this as:
```
        for {
          _ <- tosService.acceptTosStatus(userId)
          directoryEnabled <- directoryDAO.isEnabled(userId, samRequestContext)
          _ <- if (directoryEnabled) registrationDAO.enableIdentity(userId, samRequestContext) else IO.none
          status <- IO.fromFuture(IO(getUserStatus(userId, false, samRequestContext)))
        } yield status
```
I think both approaches have their merits, I don't feel strongly about changing this.

**Review by davidangb — APPROVED** [review]
optional feedback inline, else :+1:

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/service/UserService.scala (hunk 1 above)
Comment (jordansafer) [reply src/main/scala/org/broadinstitute/dsde/workbench/sam/service/UserService.scala:149]: Ah! I prefer your approach given that we're already using for-iterations- updating-

**Pull request merged**